### ISSUE TITLE
AWS097 kms key policy should only apply to IAM policies not KMS key policies

### ISSUE THREAD

**Issue by pguinardco** [tfsec/tfsec#794]
This may be a bug that's unable to be worked due to filtering in terraform, documentation may be able to be updated to reference IAM policies instead of key policies.

tfsec check AWS097 applies to all aws_iam_policy_document resources but AWS documentation has the resource of "*" required. This should be only for policies attached to IAM resources rather than KMS keys.

AWS documentation around KMS keys requiring resource of "*"

Documentation on https://tfsec.dev/docs/aws/AWS097/ requiring KMS key policies to not be on resource "*" for AWS IAM policies

https://docs.aws.amazon.com/securityhub/latest/userguide/securityhub-standards-fsbp-controls.html#fsbp-kms-1

Terraform has a circular dependency map with the suggested fix from https://tfsec.dev/docs/aws/AWS097/

Steps to reproduce the behaviour:
Run a terraform plan across the following code to produce a cycle dependency, a KMS key policy can't be attached to a KMS key policy until the KMS key is created, which requires the kms key policy:

```
resource "aws_kms_key" "key" {
  description         = "tfsec example"
  policy              = data.aws_iam_policy_document.policy.json
}

data "aws_iam_policy_document" "policy" {
  statement {
    sid = "Admins"

    effect = "Allow"

    actions = ["kms:*"]

    resources = [aws_kms_key.key.arn]

    principals {
      type        = "AWS"
      identifiers = "arn:aws:iam::${data.aws_caller_identity.current.account_id}:root"
    }
  }
}
```
Error:
```
Error: Cycle: data.aws_iam_policy_document.policy, aws_kms_key.key
```

KMS actions should apply only to IAM related policies rather than KMS related resources from https://docs.aws.amazon.com/kms/latest/developerguide/control-access-overview.html#overview-policy-elements

**Comment by owenrumney** [comment]
Further information from @aston-r in #797 

Hello,

Here is the default policy that is attached to KMS:
```
       1 | data "aws_caller_identity" "current" {}
       2 | 
       3 | data "aws_iam_policy_document" "kms_policy" {
       4 |   statement {
       5 |     sid    = "Enable IAM User Permissions"
       6 |     effect = "Allow"
       7 | 
       8 |     principals {
       9 |       identifiers = [
      10 |         "arn:aws:iam::${data.aws_caller_identity.current.account_id}:root",
      11 |       ]
      12 | 
      13 |       type = "AWS"
      14 |     }
      15 | 
      16 |     actions   = ["kms:*"]
      17 |     resources = ["*"]
      18 |   }
      19 | }

  Impact:     Identities may be able to decrypt data which they should not have access to
  Resolution: Scope down the resources of the IAM policy to specific keys

  https://tfsec.dev/docs/aws/AWS097/ 
  https://docs.aws.amazon.com/securityhub/latest/userguide/securityhub-standards-fsbp-controls.html#fsbp-kms-1
```

I had a lot of alarms regarding the policy that is attached to KMS key. According  to AWS documentation:
* https://docs.aws.amazon.com/kms/latest/developerguide/key-policies.html#key-policy-default
===
Resource – (Required) In a key policy, the value of the Resource element is "*", which means "this CMK." The asterisk ("*") identifies the CMK to which the key policy is attached.
===

* https://docs.aws.amazon.com/kms/latest/developerguide/iam-policies-best-practices.html
===
Avoid "Resource": "*" in an IAM policy. Use wildcard characters (*) judiciously. In a key policy, the wildcard character in the Resource element represents the CMK to which the key policy is attached. But in an IAM policy, a wildcard character alone in the Resource element ("Resource": "*") applies the permissions to all CMKs in all AWS accounts that the principal's account has permission to use. This might include CMKs in other AWS accounts, as well as CMKs in the principal's account.
===


To sum up: Using asterisk '\*' for resources in KMS policy that is attached to KMS is perfectly fine(wildcard character in the Resource element represents the CMK to which the key policy is attached) and all AWS examples contain "*"(see https://docs.aws.amazon.com/kms/latest/developerguide/key-policies.html )

As for the link: https://docs.aws.amazon.com/securityhub/latest/userguide/securityhub-standards-fsbp-controls.html#fsbp-kms-1 It clearly states that it relates to _IAM policies_ that are attached to _Groups/Users_, but not to KMS policies, so the check implementation is wrong.

**Comment by tinder-tder** [comment]
if you dont need cross account access you can do something like this as a work around
["arn:aws:kms:${data.aws_region.current.name}:${data.aws_caller_identity.current.account_id}:key/*"]
instead of ["*"]

**Comment by liamg** [comment]
The check no longer applies to KMS policies as of `v0.43.0` :+1:
